File: app/src/main/java/empire/stark/firststep/main/dagger/MainActivityModule.java
package empire.stark.firststep.main.dagger;

import android.app.Activity;
import android.app.Application;
import android.util.Log;

import dagger.Module;
import dagger.Provides;
import empire.stark.firststep.data.YenMinh;
import empire.stark.firststep.main.MainActivityContract;
import empire.stark.firststep.main.view.MainActivity;

/**
 * Created by YEN_MINH on 4/11/2017 3:10 AM.
 * Code and Life ( ^ .,,. ^ )
 * ---------------------------------------------
 */

@Module
public class MainActivityModule {

    @Provides
    MainActivityContract.View view(MainActivity mainActivity) {
        return mainActivity;
    }

    @Provides
    YenMinh application(Application application) {
        if (application != null) {
            Log.d("MainActivityModule", "application != null");
        } else {
            Log.e("MainActivityModule", "application == null");
        }
        return new YenMinh(application);
    }


    /*@Provides
    Activity activity(MainActivity mainActivity) {
        return mainActivity;
    }*/
}

File: app/src/main/java/empire/stark/firststep/main/dagger/MainActivityModule.kt
package empire.stark.firststep.main.dagger

import dagger.Module
import dagger.android.ContributesAndroidInjector
import empire.stark.firststep.common.dagger.BaseActivityModule
import empire.stark.firststep.common.dagger.BaseFragmentModule
import empire.stark.firststep.common.dagger.scope.PerFragment
import empire.stark.firststep.main.view.MainFragment

/**
 * Created by YEN_MINH on 4/11/2017 3:10 AM.
 * Code and Life ( ^ .,,. ^ )
 * ---------------------------------------------
 */

@Module
abstract class MainActivityModule {
    @PerFragment
    @ContributesAndroidInjector(modules = [(BaseActivityModule::class), (MainFragmentModule::class), (BaseFragmentModule::class)])
    abstract fun mainFragment(): MainFragment
}
